pvlib/iotools/psm3.py

```diff
@@ -0,0 +1,133 @@
+"""
+Get PSM3 TMY
+see https://developer.nrel.gov/docs/solar/nsrdb/psm3_data_download/
+"""
+
+import io
+import csv
+import datetime
+import requests
+import pandas as pd
+import pytz
+
+URL = "http://developer.nrel.gov/api/solar/nsrdb_psm3_download.csv"
+
+# 'relative_humidity', 'total_precipitable_water' are note available
+ATTRIBUTES = [
+    'air_temperature', 'dew_point', 'dhi', 'dni', 'ghi', 'surface_albedo',
+    'surface_pressure', 'wind_direction', 'wind_speed']
+
+
+def get_psm3(latitude, longitude, tmy='tmy', interval=60):
+    """
+    Get PSM3 data
+
+    Parameters
+    ----------
+    latitude : float or int
+        in decimal degrees, between -90 and 90, north is positive
+    longitude : float or int
+        in decimal degrees, between -180 and 180, east is positive
+    tmy : str
+        PSM3 parameter "name", see notes below for options, default ``'tmy'``
+    interval : integer
+        interval size in minutes, can be only either 30 or 60, default 60
+
+    Returns
+    -------
+    headers : dict
+        meta data from NREL PSM-3 about the record, see notes for fields
+    data : pandas.DataFrame
+        timeseries data from NREL PSM3
+
+    Raises
+    ------
+    requests.HTTPError
+
+    Notes
+    -----
+    The ``tmy`` parameter 'names' must be a single value from the following
+    list:
+
+        ['1998', '1999', '2000', '2001', '2002', '2003', '2004', '2005',
+         '2006', '2007', '2008', '2009', '2010', '2011', '2012', '2013',
+         '2014', '2015', '2016', '2017', 'tmy', 'tmy-2016', 'tmy-2017',
```

Comment:
`tmy` and variants do not appear to be options on the API documentation linked to above. Did I miss it?

Reply:
IMO their documentation is spotty.  `'tmy'` definitely works I haven't tried the variants.  There's an example of downloading one year [here](https://nsrdb.nrel.gov/api-instructions) the docs do say that only one year can be downloaded this way but I haven't tested that.  The example says zenith can be downloaded,  but in tests I couldn't get precipitable water or relative humidity. This api could use more rigorous testing due to the spotty documentation IMO

Reply:
Did you try the variants? They don't need to be automatically tested, but I'd like to know that someone checked that they worked.

Reply:
no I didn't try the variants ...

Reply:
`tmy-2016` works
`2016` works